sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -57,20 +59,20 @@
 
   // The data stored in these two allocations need to maintain binary compatible. We can
   // directly pass this buffer to external components.
-  private long nulls;
```

Comment:
do we have to change this class? column vector doesn't allocate memory from the memory manager.

Reply:
I think that we use `MemoryBlock` for this column vector in `reserveInternal()`. The method `reallocate()` returns `MemoryBlock`.
`this.nulls = MemoryAllocator.UNSAFE.reallocate(nulls, oldCapacity, newCapacity);`

Reply:
what if we don't remove `Platform.reallocateMemory`?

Reply:
Ah, you want to allocate memory for `OffHeapColumnVector` by using `Platform` instead of `MemoryBlock`?
Could you please explain why  `OffHeapColumnVector`  wants to allocate memory `Platform`?

Reply:
To remove `Platform.reallocateMemory` is not a strong reason to migrate `OffHeapColumnVectot` to memory block, we can do it later, and update `OnHeapColumnVector` too.

Reply:
I see. `Platform.reallocateMemory` does not exist.
`MemoryAllocator.UNSAFE.reallocate()` returns `MemoryBlock`.

Reply:
What I mean is, we don't need to change `OffHeapColumnVectot` at all, if we don't remove `Platform.reallocateMemory`. We should apply the new `MemoryBlock` incrementally. I think applying it to `UTF8String` is a good start to prove the performance advantage, but I don't think we need to rush to column vectors in this PR.

Reply:
I agree with incremental updates. Do we revert changes in `OffHeapColumnVector` in this PR? In other words, we do not change `OffHeapColumnVector` at all in this PR.

Reply:
yea, I think `UTF8String` is good enough as the first show case.

Reply:
I see. I will revert changes in 'OffHeapColumnVector' tonight.

Reply:
@cloud-fan  Reverted.